src/AwsClient.php

```diff
@@ -407,45 +414,37 @@ private function addSignatureMiddleware()
     {
         $api = $this->getApi();
         $provider = $this->signatureProvider;
-        $version = $this->config['signature_version'];
+        $signatureVersion = $this->config['signature_version'];
         $name = $this->config['signing_name'];
         $region = $this->config['signing_region'];
 
         $resolver = static function (
             CommandInterface $c
-        ) use ($api, $provider, $name, $region, $version) {
+        ) use ($api, $provider, $name, $region, $signatureVersion) {
             if (!empty($c['@context']['signing_region'])) {
                 $region = $c['@context']['signing_region'];
             }
             if (!empty($c['@context']['signing_service'])) {
                 $name = $c['@context']['signing_service'];
             }
+            if (!empty($c['@context']['signature_version'])) {
+                $signatureVersion = $c['@context']['signature_version'];
+            }
 
             $authType = $api->getOperation($c->getName())['authtype'];
-            switch ($authType){
+            switch ($authType) {
                 case 'none':
-                    $version = 'anonymous';
+                    $signatureVersion = 'anonymous';
                     break;
                 case 'v4-unsigned-body':
-                    $version = 'v4-unsigned-body';
+                    $signatureVersion = 'v4-unsigned-body';
                     break;
                 case 'bearer':
                     $version = 'bearer';
```

Comment:
Shouldn't this be `signatureVersion` now? Is there a test set for this?

Reply:
There wasn't a test on this when bearer was implemented. I've added a case in `AwsClientTest` (`testSignOperationsWithAnAuthType`) which covers this.